modules/airavata-helix/task-core/src/main/java/org/apache/airavata/helix/core/participant/HelixParticipant.java

```diff
@@ -178,6 +189,26 @@ private void connect() {
 
     private void disconnect() {
         if (zkHelixManager != null) {
+            if (runningTasks.size() > 0) {
+                for (int i = 0; i <= PARTICIPANT_SHUTDOWN_GRACE_RETRIES; i++) {
+                    logger.info("Shutting down gracefully [RETRY " + i + "]");
+                    try {
+                        Thread.sleep(PARTICIPANT_SHUTDOWN_GRACE_PERIOD);
+                    } catch (InterruptedException e) {
+                        logger.warn("Waiting for running tasks failed [RETRY " + i + "]", e);
+                    }
+                    if (runningTasks.size() == 0) {
+                        break;
+                    }
+                }
+
+                synchronized (runningTasks) {
+                    for (AbstractTask runningTask : runningTasks) {
+                        logger.warn("Cancelling task with id: " + runningTask.getTaskId() + " forcefully");
+                        runningTask.cancel();
```

Comment:
Calling cancel method like this will not actually cancel the task. You have to do this through task driver.